intellst-api/src/Service/Register.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\User;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Doctrine\ORM\getDoctrine;
+
+class Register extends AbstractController
+{
+    public function __construct(RequestStack $requestStack, UserPasswordEncoderInterface $encoder)
+    {
+        $this->requestStack = $requestStack;
+        $this->encoder = $encoder;
+    }
+
+    public function register() : string
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $request = $this->requestStack->getCurrentRequest();
+        $decoder = json_decode($request->getContent(), true);
```

Comment:
you should use serializer to decode directly to a DTO object, then transformer to transform DTO to User entity

Reply:
this i will do in the next tasc registration